miniconf.py

```diff
@@ -51,7 +52,8 @@ def _handle_response(self, _client, topic, payload, *_args, **_kwargs):
             logger.warning('Unexpected response on topic: %s', topic)
             return
 
-        self.inflight[topic].set_result(payload.decode('ascii'))
+        response = json.loads(payload)
+        self.inflight[topic].set_result((response['code'], response['msg']))
```

Comment:
Because tuples are generally much more accessible as return values than dictionaries. If you use a dictionary as a return argument, the end user then needs to know the specific keys to use to access the underlying data. However, if we use a tuple, it allows them to directly unpack like so:
```python
return_code, message = await future
```

Without unpacking into a tuple, it would have to be (note that the user now needs to know about the specific keys):
````python
response = await future
code = response['code']
message = response['msg']
````

Reply:
In fact, not unpacking it at the API layer actually _forces_ us to document the response structure and make it backward compatible. With as-implemented, we can change the entire reporting structure of miniconf without worrying about breaking the external API.